nifi-extension-bundles/nifi-network-bundle/nifi-network-processors/src/main/java/org/apache/nifi/processors/network/util/PCAP.java

```diff
@@ -0,0 +1,717 @@
+            } else {
+                this.versionMajor = this.io.readU2le();
+                if (!(versionMajor() == 2)) {
+                    throw new ByteBufferInterface.ValidationNotEqualError("Packet major version is not 2.");
+                }
+                this.versionMinor = this.io.readU2le();
+                this.thiszone = this.io.readS4le();
+                this.sigfigs = this.io.readU4le();
+                this.snaplen = this.io.readU4le();
+                this.network = PCAP.Linktype.byId(this.io.readU4le());
+            }
+        }
+
+        private byte[] magicNumber;
+        private int versionMajor;
+        private int versionMinor;
+        private int thiszone;
+        private long sigfigs;
+        private long snaplen;
+        private Linktype network;
+        private PCAP root;
+        private PCAP parent;
+
+        public byte[] magicNumber() {
+            return magicNumber;
+        }
+
+        public int versionMajor() {
+            return versionMajor;
+        }
+
+        public int versionMinor() {
+            return versionMinor;
+        }
+
+        /**
+         * Correction time in seconds between UTC and the local
+         * timezone of the following packet header timestamps.
+         */
+        public int thiszone() {
+            return thiszone;
+        }
+
+        /**
+         * In theory, the accuracy of time stamps in the capture; in
+         * practice, all tools set it to 0.
+         */
+        public long sigfigs() {
+            return sigfigs;
+        }
+
+        /**
+         * The "snapshot length" for the capture (typically 65535 or
+         * even more, but might be limited by the user), see: incl_len
+         * vs. orig_len.
+         */
+        public long snaplen() {
+            return snaplen;
+        }
+
+        /**
+         * Link-layer header type, specifying the type of headers at
+         * the beginning of the packet.
+         */
+        public Linktype network() {
+            return network;
+        }
+
+        public PCAP root() {
+            return root;
+        }
+
+        public PCAP parent() {
+            return parent;
+        }
+    }
+
+    /**
+     * @see <a href=
+     *      "https://wiki.wireshark.org/Development/LibpcapFileFormat#Record_.28Packet.29_Header">Source</a>
+     */
+    public static class Packet {
+        public ByteBufferInterface io;
+
+        public Packet(ByteBufferInterface io) {
+            this(io, null, null);
+        }
+
+        public Packet(ByteBufferInterface io, PCAP parent) {
+            this(io, parent, null);
+        }
+
+        public Packet(ByteBufferInterface io, PCAP parent, PCAP root) {
+
+            this.parent = parent;
+            this.root = root;
+            this.io = io;
+            read();
+        }
+
+        public Packet(long tSSec, long tSUsec, long inclLen, long origLen, byte[] rawBody, String network) {
```

Comment:
Intellij says `network` is never used
```suggestion
        public Packet(long tSSec, long tSUsec, long inclLen, long origLen, byte[] rawBody) {
```